lib/irb.rb

```diff
@@ -595,13 +595,30 @@ def each_top_level_statement
       end
     end
 
+    SIMPLE_COMMAND_REGEXP = /^(?<cmd_name>\S+)$/
+    COMMAND_WITH_ARGS_REGEXP = /^(?<cmd_name>\S+) +(?<cmd_arg>[^-]\S*)$/
+    COMMAND_WITH_FLAGS_REGEXP = /^(?<cmd_name>\S+) +(?<cmd_flag>-\S+( +\S+)*)$/
+    COMMAND_WITH_ARGS_AND_FLAGS_REGEXP = /^(?<cmd_name>\S+) +(?<cmd_arg>[^-]\S*) +(?<cmd_flag>-\S+( +\S+)*)$/
+
+    COMMAND_REGEXP = Regexp.union(
+      SIMPLE_COMMAND_REGEXP,
+      COMMAND_WITH_ARGS_REGEXP,
+      COMMAND_WITH_FLAGS_REGEXP,
+      COMMAND_WITH_ARGS_AND_FLAGS_REGEXP
+    )
+
     def build_statement(code)
       code.force_encoding(@context.io.encoding)
-      command_or_alias, arg = code.split(/\s/, 2)
-      # Transform a non-identifier alias (@, $) or keywords (next, break)
-      command_name = @context.command_aliases[command_or_alias.to_sym]
-      command = command_name || command_or_alias
-      command_class = ExtendCommandBundle.load_command(command)
+      command_match = COMMAND_REGEXP.match(code.strip)
+
+      if command_match
+        command_or_alias = command_match[:cmd_name]
+        arg = [command_match[:cmd_arg], command_match[:cmd_flag]].compact.join(' ')
```

Comment:
I will open a follow up PR to see if extracting `cmd_flag` can simplify how commands implement options.